src/index.ts

```diff
@@ -1,13 +1,11 @@
-import * as extend from 'extend';
-import * as _request from 'request';  // for types only
-import * as request from 'retry-request';
+import axios, {AxiosError, AxiosRequestConfig, AxiosResponse} from 'axios';
 
 const BASE_URL = 'http://metadata.google.internal/computeMetadata/v1';
 
-export type Options = _request.CoreOptions&{property?: string, uri?: string};
+export type Options = AxiosRequestConfig&{property?: string, uri?: string};
```

Comment:
This is a semver major change. To aid in migration it would be good to throw when `request` options are passed in by accident; e.g. throw on `qs`.

Reply:
README.md has an example that shows `qs`. That example will be invalid after this change, so it need to be edited as part of this PR.

Reply:
👍 fixed in both